python/sanlock.c

```diff
@@ -136,14 +190,16 @@ __parse_resource(PyObject *obj, struct sanlk_resource **res_ret)
 
         }
 
-        if (!PyArg_ParseTuple(disk, "sK", &path, &offset)) {
+        if (!PyArg_ParseTuple(disk, "O&K", pypath_converter, &path, &offset)) {
             /* Override the error since it confusing in this context. */
             set_error(PyExc_ValueError, "Invalid disk %s", disk);
+            Py_XDECREF(path);
```

Comment:
Why cleanup here?

Reply:
the patch here is inside a for loop over resource disks (path, offset)
so we need to decref path object as soon as we can before next iteration, where we parse another path and loose the previous reference and cause a leak.

Reply:
Make sense, but in this case path should be defined in the scope of the loop, and cleaned up inside the loop.